examples/chat.rs

```diff
@@ -409,19 +415,61 @@ impl Client {
     }
 
     fn handle_media_added(&mut self, mid: Mid, kind: MediaKind) -> Propagated {
-        let track_in = Arc::new(TrackIn {
-            origin: self.id,
-            mid,
-            kind,
-        });
+        let track_in = TrackInEntry {
+            id: Arc::new(TrackIn {
+                origin: self.id,
+                mid,
+                kind,
+            }),
+            last_keyframe_request: None,
+        };
 
         // The Client instance owns the strong reference to the incoming
         // track, all other clients have a weak reference.
-        let weak = Arc::downgrade(&track_in);
+        let weak = Arc::downgrade(&track_in.id);
         self.tracks_in.push(track_in);
 
         Propagated::TrackOpen(self.id, weak)
     }
+    
+    fn handle_media_data_in(&mut self, data: MediaData) -> Propagated {
+        if !data.contiguous {
+            self.request_keyframe_throttled(
+                data.mid,
+                data.rid,
+                KeyframeRequestKind::Fir,
+            );
+        }
```

Comment:
If we are requesting a keyframe, should we really propagate the data?

Reply:
I think so, with VP9 and VP8 for example, it's possible that the decoder can still do something reasonable because of temporal scalability. Also, if we don't send anything then we'll trigger `onmute` on the `MediaStreamTrack` instead

Reply:
Isn't that the wishful thinking we had in our internal service (derp)?

When using media level, the remote side can't see the discontinuity in RTP sequence, so we're effectively hoping the VP9 packet is enough for it to "do something reasonable" with it?

I know that both VP8 and VP9 has an internal sequence number, but didn't we stop doing this for derp?

Reply:
We still forward, in addition to requesting a PLI